src/BlazorWasmSdk/Targets/Microsoft.NET.Sdk.BlazorWebAssembly.6_0.targets

```diff
@@ -63,25 +60,58 @@ Copyright (c) .NET Foundation. All rights reserved.
 
   <!-- Wire-up static web assets -->
   <PropertyGroup>
-    <GenerateStaticWebAssetsPublishManifestDependsOn Condition="'$(WasmBuildingForNestedPublish)' != 'true'">
-      $(GenerateStaticWebAssetsPublishManifestDependsOn);
-      _BlazorCompressPublishFiles;
-    </GenerateStaticWebAssetsPublishManifestDependsOn>
+    <ResolveCompressedFilesDependsOn>
+      $(ResolveCompressedFilesDependsOn);
+      _ResolveBlazorWasmOutputs;
+    </ResolveCompressedFilesDependsOn>
+
+    <ResolveCompressedFilesForPublishDependsOn>
+      $(ResolveCompressedFilesForPublishDependsOn);
+      _CollectAssetsToCompressForPublish;
+    </ResolveCompressedFilesForPublishDependsOn>
 
     <AddWasmStaticWebAssetsDependsOn>
       $(AddWasmStaticWebAssetsDependsOn);
       _ResolveBlazorWasmOutputs;
     </AddWasmStaticWebAssetsDependsOn>
+
     <GenerateBuildWasmBootJsonDependsOn>
       $(GenerateBuildWasmBootJsonDependsOn);
       GenerateBlazorBootExtensionJson;
     </GenerateBuildWasmBootJsonDependsOn>
+
     <GeneratePublishWasmBootJsonDependsOn>
       $(GeneratePublishWasmBootJsonDependsOn);
       GenerateBlazorBootExtensionJson;
     </GeneratePublishWasmBootJsonDependsOn>
   </PropertyGroup>
 
+  <ItemGroup>
+
+    <!--
+      Compress referenced binaries using GZip during build. This skips files such as the project's assemblies
+      that change from build to build. Runtime assets contribute to the bulk of the download size. Compressing it
+      has the most benefit while avoiding any ongoing costs to the dev inner loop.
+    -->
+    <CompressionConfiguration Include="BlazorBuildGzip">
+      <Stage>Build</Stage>
+      <Format>gzip</Format>
+      <IncludePattern>_framework\**</IncludePattern>
+      <ExcludePattern>_framework\blazor.boot.json</ExcludePattern>
+    </CompressionConfiguration>
+
+    <CompressionConfiguration Include="BlazorPublishGzip">
+      <Stage>Publish</Stage>
+      <Format>gzip</Format>
+      <TargetDirectory>$(_BlazorPublishCompressDirectory)</TargetDirectory>
+    </CompressionConfiguration>
+    <CompressionConfiguration Include="BlazorPublishBrotli">
+      <Stage>Publish</Stage>
+      <Format>brotli</Format>
+    </CompressionConfiguration>
+
+  </ItemGroup>
+
```

Comment:
I think this should be inside the wasm SDK, and not inside the Blazor specific bits (since all the runtime functionality moved there)